Display cross link mw topology

abonas opened the pull request:
@miq-bot add_label wip, topology, providers/hawkular

![crosslinktopo](https://cloud.githubusercontent.com/assets/6277245/17477209/997e005e-5d6c-11e6-9816-e93f3b4b1e5b.png)

abonas commented:
note: this PR will change after #8630 is merged. marked as wip for now, but works.
@theute

abonas commented:
@miq-bot rm_label wip

abonas commented:
@miq-bot assign @Jiri-Kremser 
@Jiri-Kremser please review. in particular I'm concern with the getCircleDimensions method that requires x,y,r,height, width although not all are necessarily relevant

Jiri-Kremser commented on app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js:
```diff
@@ -141,18 +141,22 @@ function MiddlewareTopologyCtrl($scope, $http, $interval, $location, topologySer
+                return { x: defaultDimensions.x, y: defaultDimensions.y, r: 21 };
             default :
                 return { x: -9, y: -9, height: 18, width: 18, r: 17 };
         }
```
shouldn't the default case use the `defaultDimensions`?

abonas replied:
that's part of my concerns. the dimensions part changed because the code now mixes dimensions of font stuff and png. and so it is less suitable as it is now. (and the controller sets some dimensions [hard coded](https://github.com/ManageIQ/manageiq/blob/master/app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js#L116))

abonas commented:
@miq-bot add_label wip

abonas commented:
@miq-bot rm_label wip

abonas commented:
@Jiri-Kremser I rebased and fixed couple of things, please review.
The default case as I said above is trickier now because of mixing font and image.
But I added the height/weight to default dimensions in topology service and they are reused in mw topology.

abonas commented:
> @Jiri-Kremser I rebased and fixed couple of things, please review.
> The default case as I said above is trickier now because of mixing font and image.
> But I added the height/weight to default dimensions in topology service and they are reused in mw topology.

I removed the height/weight from default dimensions in the generic service as it "made noise" and failed tests in other topologies. I think it should be readded in another PR that is not focused on cross link in middleware.

abonas commented:
@miq-bot add_label enhancement

Jiri-Kremser commented:
lgtm

abonas commented:
@miq-bot assign @chessbyte

abonas commented:
@dclarizio can someone review this and merge please? it's not a big PR.

abonas commented:
@martinpovolny @himdel maybe there's a chance you could have a look/merge this one? thanks.
